ui/plugin-system/src/model/time-series-queries.ts

```diff
@@ -41,6 +41,18 @@ export interface TimeSeriesData {
 export interface TimeSeries {
   name: string;
   values: Iterable<TimeSeriesValueTuple>;
+  formattedName?: string;
 }
 
 export type TimeSeriesValueTuple = [timestamp: UnixTimeMs, value: number];
+
+export type SeriesLabels = Record<string, string>;
+
+/*
+ * Formatter used for series name display in legends and tooltips
+ * Regex replaces label {{ name }} with resolved label value
+ */
+export function formatSeriesName(inputFormat: string, seriesLabels: SeriesLabels): string {
```

Comment:
I added this function in plugin-system since it isn't specific to PrometheusTimeSeriesQuery, but interested in feedback on where exactly it should go (core is another option). 

Also, this regex can be updated pretty easily if we want to go with something different than the handlebars `{{example}}` syntax, any alternatives that people prefer?

Reply:
Yeah that makes sense about keeping here, but moving it out of model folder--we due sometimes put functions in model files, but agreed this one makes more sense in a `utils` folder. 

Also, @LukeTillman if we decide to go with the potential new `runtime` lib would this formatSeriesName function go there?

Reply:
🤔  Probably not? I think it would probably stay here. I was thinking `runtime` would be more for things that are related to the "runtime state of the app", but definitely something to think about some more.

Reply:
I scrolled down and see that you put this option into the Prometheus-specific types. If that's the case, this should _probably_ live in the `prometheus-plugin` package for now. If we move the option to the spec for ALL time series queries, regardless of plugin, it should probably live here. If we want it to just be a "utility" function that individual plugins could opt into using in their own options, we could move it to `core`.

Reply:
Okay that makes sense about keeping it consistent! I ended up moving it to utils.ts in ui/prometheus-plugin

It's possible it could go up a level on `kind: 'TimeSeriesQuery'`, the regex in the `formatSeriesName` function isn't really specific to Prom, but it wasn't clear to me how to make that happen as far as the editor components go? Maybe something we can follow up on